feat: implement cascading model fallback on 408 timeout

## Problem

Users experience 408 Request Timeout errors with no progress when using slower models (Opus) with large context or slow tools. The current retry logic retries with the same model, which often fails again.

From console logs:
```
[ERROR] HTTP 408 Request Timeout from Gateway
[INFO] Retrying explicitly-selected model anthropic/claude-opus-4.5 (attempt 1/1)
[WARN] Transient error... after 1 retry, giving up
```

## Proposed Solution

Implement a three-tier fallback strategy:

### 1. Cascading Model Fallback (Primary - Rust Backend)
Automatically downgrade to faster models on 408 timeout:
- Opus (408) → Auto-retry with Sonnet
- Sonnet (408) → Auto-retry with Haiku  
- Haiku (408) → Show error (nothing faster)

**Changes needed:**
- Modify orchestrator retry logic in `src-tauri/src/orchestrator/`
- Add model fallback chain to retry strategy
- Preserve conversation context across model switches

### 2. User Notification (Secondary - Frontend)
Show toast notification when model fallback occurs:
- "⚡ Switched to Claude Sonnet for faster response"
- Include reason (timeout) and new model name
- Don't interrupt the conversation flow

**Changes needed:**
- Add toast notification component (if not exists)
- Wire orchestrator model-switch event to frontend
- Display notification without blocking UX

### 3. Proactive Model Selection (Tertiary - Prevention)
Detect large context and proactively use faster model:
- Monitor conversation token count
- Auto-downgrade when context > 50k tokens
- Show notification: "Switched to Sonnet for large context"

**Changes needed:**
- Add token counting in conversation store
- Implement `selectOptimalModel()` function
- Apply before sending each prompt

## Success Criteria

- ✅ 408 timeouts automatically retry with faster model
- ✅ Users see progress instead of errors
- ✅ Conversation continues seamlessly
- ✅ Users are informed when model switches occur
- ✅ Large contexts automatically use faster models

## Technical Notes

- Preserve user's explicit model selection for future prompts
- Only fallback for the current prompt that timed out
- Reset to user-selected model after successful completion
- Log all model switches for debugging

Taariq Lewis, SerenAI, Paloma, and Volume at https://serendb.com
Email: hello@serendb.com

--- a/src-tauri/src/orchestrator/service.rs
+++ b/src-tauri/src/orchestrator/service.rs
@@ -380,8 +380,9 @@ async fn execute_single_task(
 
             // Try cascading to a faster model on timeout (Opus→Sonnet→Haiku)
             if is_timeout_error {
-                if let Some(fallback_model) = get_fallback_model(&routing.model_id) {
+                if let Some(fallback_model_str) = get_fallback_model(&routing.model_id) {
                     let failed_model = routing.model_id.clone();
+                    let fallback_model = fallback_model_str.to_string(); // Convert to owned String
 
                     log::info!(
                         "[Orchestrator] 408 timeout on {}, falling back to faster model: {}",
@@ -394,7 +395,7 @@ async fn execute_single_task(
                         conversation_id: conversation_id.to_string(),
                         worker_event: WorkerEvent::Reroute {
                             from_model: failed_model.clone(),
-                            to_model: fallback_model.to_string(),
+                            to_model: fallback_model.clone(),
                             reason: "Switched to faster model due to timeout".to_string(),
                         },
                         subtask_id: None,
@@ -402,8 +403,8 @@ async fn execute_single_task(
                     let _ = app.emit("orchestrator://event", &reroute_event);
 
                     // Update routing to use fallback model
-                    routing.model_id = fallback_model.to_string();
-                    tried_models.push(fallback_model.to_string());
+                    routing.model_id = fallback_model.clone();
+                    tried_models.push(fallback_model);
 
                     // Brief backoff before retry with new model
                     tokio::time::sleep(std::time::Duration::from_secs(2)).await;

